mem0/memory/main.py

```diff
@@ -1483,7 +1490,7 @@ async def _create_procedural_memory(self, messages, metadata=None, llm=None, pro
 
         return result
 
-    async def _update_memory(self, memory_id, data, existing_embeddings, metadata=None):
+    async def _update_memory(self, memory_id, data, existing_embeddings, metadata: dict | None = None):
```

Comment:
Change this to `Optional[dict]` because this project supports Python >=3.9 and `|` was introduced in Python 3.10